alien4cloud-yorc-plugin/src/main/java/org/ystia/yorc/alien4cloud/plugin/EventListenerTask.java

```diff
@@ -257,6 +254,39 @@ public void run() {
                                             break;
                                     }
                                     break;
+                                case EVT_ATTRIBUTE_CHANGE:
+                                    eNode = event.getNodeId();
+                                    eInstance = event.getInstanceId();
+                                    String eAttribute = event.getAttribute();
+                                    String eValue = event.getValue();
+                                    eMessage += event.getType() + ":" + eState + "";
+                                    String.format("Attribute (%s) updated to value:%s for node:%s, instance:%s", eAttribute, eValue, eNode, eInstance);
+                                    log.debug("Received Event from Yorc <<< " + eMessage);
+
+                                    ninfo = instanceInfo.get(eNode);
+                                    if (ninfo == null) {
+                                        // Add a new Node in YorcRuntimeDeploymentInfo
+                                        log.debug("Add a node in YorcRuntimeDeploymentInfo: " + eNode);
+                                        ninfo = Maps.newHashMap();
+                                        instanceInfo.put(eNode, ninfo);
+                                    }
+                                    iinfo = ninfo.get(eInstance);
+                                    if (iinfo == null) {
+                                        // Add a new Instance for this node in YorcRuntimeDeploymentInfo
+                                        log.debug("Add an instance in YorcRuntimeDeploymentInfo: " + eInstance);
+                                        iinfo = orchestrator.newInstance(new Integer(eInstance));
+                                        ninfo.put(eInstance, iinfo);
+                                    }
+
+                                    iinfo.getAttributes().put(eAttribute, eValue);
+
+                                    // Notify a4c
+                                    PaaSInstanceStateMonitorEvent a4cEvent = new PaaSInstanceStateMonitorEvent();
+                                    a4cEvent.setInstanceId(eInstance);
+                                    a4cEvent.setNodeTemplateId(eNode);
+                                    a4cEvent.setAttributes(iinfo.getAttributes());
```

Comment:
The instanceState is not initialized here.
But Alien4Cloud when it receives a PaaSInstanceStateMonitorEvent with an uninitialized instanceState considers this is an 'instance deleted' event : https://github.com/alien4cloud/alien4cloud/blob/develop/alien4cloud-ui/src/main/webapp/scripts/applications/services/runtime_event_service.js#L79

So there are a lot of unexpected events in Alien4Cloud UI `Instance myinstance/0 has been removed`

Example of a sequence:
```
12/2/19 14:24:13 | Instance status | Instance myinstance/0 changed status to started
12/2/19 14:24:13 | Instance status | Instance myinstance/0 has been removed
12/2/19 14:24:13 | Instance status | Instance myinstance/0 has been removed
12/2/19 14:24:13 | Instance status | Instance myinstance/0 has been removed
12/2/19 14:24:13 | Instance status | Instance myinstance/0 has been removed
12/2/19 14:24:13 | Instance status | Instance myinstance/0 has been removed
12/2/19 14:23:43 | Instance status | Instance myinstance/0 has been removed
12/2/19 14:23:42 | Instance status | Instance myinstance/0 changed status to creating
```
The instanceState should be set to avoid this issue.

Reply:
just fixed